### PR TITLE
fix issue 15409 - Mallocator cant be used in @nogc code

### PR DESCRIPTION
**Pull request by BBasile**
note that no additional unittest is needed since I annotated those which already exist with @nogc to force the checking.

### CODE REVIEW

File: std/experimental/allocator/mallocator.d [hunk 1]
```diff
@@ -64,19 +64,19 @@ struct Mallocator
...
 
-unittest
+@nogc unittest
 {
-    static void test(A)()
+    @nogc static void test(A)()
     {
         int* p = null;
-        p = new int;
+        p = cast(int*) A.instance.allocate(int.sizeof);
```
Comment (JackStouffer) [inline std/experimental/allocator/mallocator.d]: I haven't had the chance to play with std.allocator yet, but this leaks memory if you don't call deallocate right? Why not add a `scope(exit)` to do that?

Comment (JakobOvrum) [inline std/experimental/allocator/mallocator.d]: These `test` functions appear to have been left in a state of flux. The one in the unittest below this one still appears completely nonsensical.

Comment (BBasile) [inline std/experimental/allocator/mallocator.d]: The leak doesn't matter, it'a test, so single shot program. However I can add the deallocation code if you wish to. Tell me clearly yes / no.

And yes, the test funct was initially nothing bu useless, I just modified it to get an additional @nogc checking.

Comment (JackStouffer) [inline std/experimental/allocator/mallocator.d]: It's a one line fix and there's no reason to leak unless absolutely necessary, so please add it.

**Comment by BBasile** [comment]
PR modified.

**Comment by JackStouffer** [comment]
LGTM

**Comment by Hackerpilot** [comment]
@BBasile I just approved you in the auto-tester.

**Comment by Hackerpilot** [comment]
Auto-merge toggled on

**Pull request merged**